aea/aea.py

```diff
@@ -53,9 +55,10 @@ def __init__(
         resources: Resources,
         loop: Optional[AbstractEventLoop] = None,
         timeout: float = 0.0,
+        execution_timeout: float = 0.1,
```

Comment:
TL;DR: what about `execution_timeout = 0.0` by default? Or a high value like `5.0` .

Thought: in `asyncio` programming, the library does not tell you that coroutine should be fast because otherwise they block the entire loop. Of course that is not part of "idiomatic asyncio programming" , but gives to the developer more flexibility (if I want to call `time.sleep(1.0)` in a coroutine I expect it to work despite it will block the loop).

The analogy between asyncio and the framework introduced in this PR is not strong since `aea` is not a standard library. However, the resulting experience of the developer might be worse because the `handle/act` always get blocked.

I see `execution_timeout` as a configuration for "production-ready" AEAs, whereas during development I would go for `execution_timeout=0.0` or setting it to a high value.

Also, many skills currently block the loop because of some interaction with the Decision Maker (@DavidMinarsch can you confirm?), so a default value too low or different than `0.0` would break those skills.

Reply:
@marcofavorito , last point first, interactions with the decision maker should be non-blocking (send a message to DM, pick it up via handler). Some skills might be affected though due to poor logic and it is something we need to test.

I think especially for development it would be helpful to know when you write blocking logic. But I see your point of view. Maybe we make the execution timeout configurable via the builder already?

Reply:
@marcofavorito 
python doc says: https://docs.python.org/3/library/asyncio-dev.html#running-blocking-code
so it assumed to not put a blocking code in a coroutine. anyway, a developer is free to do this.

the main idea of the execution timeout feature is to notify the developer that something wrong with his code and there can be side effects with performance.

actually we can warn it more gently, without interruption. just write a log.


in short: main proposal is just to make harder to write bad code. but it does not help to write a good one.

Reply:
yes, there's room for improvement to customize the actual behaviour of this feature (e.g. only send a notification or raise an exception). This could be the jobby a later PR. 

I wonder what's a good default value for this feature until the other PR on customizability is in place.

Reply:
better and more flexible config support is required. currently, the agent's timeout is not adjustable via config, so better agent build with config support is welcome

Reply:
updated default timeout to 1 second.
suppose it's better to make a dedicated discussion for better control over code execution and possible ways to solve it